service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java

```diff
@@ -232,6 +232,18 @@ private void processCredentialsMessageRemoveOperation(final Message<JsonObject>
         }
     }
 
+    /**
+     * Override the following method to extend RegistrationService with implementation
+     * specific operations.
+     *
+     * @param credMsg target credential message to be processed.
+     * @param action implementation specific action to be executed.
+     */
+    public void processOtherCredentialMessage(final Message<JsonObject> credMsg, String action) {
```

Comment:
Can we rename this to e.g. *processCustomCredentialsMessage*? FMPOV it should be declared *protected* instead of *public* since it is supposed to be invoked by *processCredentialsMessage* only, isn't it?

Reply:
Yes, I can rename to `processCustomCredentialsMessage`. But if custom implementation has to override it, it has to be public, right?

Reply:
I even thought of making `processCustomCredentialsMessage` abstract, but it will break existing implementations. So went for a public method with default implementation.

Reply:
I don't think so. See https://stackoverflow.com/a/11626042

Reply:
Oh..yes we are just invoking from within and of course the concrete implementation can override it.